src/pages/index.js

```diff
@@ -117,48 +114,120 @@ function getCardElement(data) {
   const cardLikeButton = cardElement.querySelector(".card__like-btn");
   const cardBinButton = cardElement.querySelector(".card__bin-btn");
 
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-btn_liked");
+  }
+
   cardImageElement.src = data.link;
   cardImageElement.alt = data.alt;
   cardNameElement.textContent = data.name;
 
-  cardLikeButton.addEventListener("click", () => {
-    cardLikeButton.classList.toggle("card__like-btn_liked");
-  });
-
-  cardBinButton.addEventListener("click", () => {
-    cardElement.remove("disabled");
-  });
-
-  cardImageElement.addEventListener("click", () => {
-    openModal(previewModal);
-    previewModalImageElement.src = data.link;
-    previewModalImageElement.alt = data.alt;
-    previewModalCaptionElement.textContent = data.name;
-  });
+  cardLikeButton.addEventListener("click", (evt) => handleLike(evt, data._id));
+  cardBinButton.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+  cardImageElement.addEventListener("click", () => handleImageClick(data));
 
   return cardElement;
 }
 
 // Event handlers
-function handleEditProfileSubmit(evt) {
+function handleAvatarSubmit(evt) {
   evt.preventDefault();
 
-  profileName.textContent = nameInput.value;
-  profileDescription.textContent = descriptionInput.value;
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileImage.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(setButtonText(submitBtn, false));
+  avatarModalForm.reset();
```

Comment:
`avatarModalForm.reset();` should be in `then`


You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.